src/common/transformations/src/transformations/common_optimizations/gru_cell_fusion.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/gru_cell_fusion.hpp"
+
+#include <memory>
+
+#include "itt.hpp"
+#include "node_registry.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+using namespace std;
+using namespace ov::element;
+using namespace ov::pass;
+using namespace ov::opset9;
+using namespace ov::pass::pattern;
+
+ov::pass::GRUCellFusion::GRUCellFusion() {
+    MATCHER_SCOPE(GRUCellFusion);
+
+    // we can't determine hidden_size or input_size in this case
+    const auto is_first_dim_dynamic = [](const Output<Node>& output) -> bool {
+        const auto& p_shape = output.get_partial_shape();
+        return !(p_shape.rank().is_dynamic() || p_shape[1].is_dynamic());
+    };
+
+    auto concat_1 = wrap_type<Concat>({any_input(is_first_dim_dynamic), any_input(is_first_dim_dynamic)});
+    auto matmul_1 = wrap_type<MatMul>({concat_1, any_input()});
```

Comment:
```suggestion
    auto matmul_1 = wrap_type<MatMul>({concat_1, any_input(is_first_dim_dynamic)});
```
Can be used here and for matmul_2 also?

Reply:
I will add additional checks

Reply:
done